sql/core/src/test/java/test/org/apache/spark/sql/JavaDatasetSuite.java

```diff
@@ -390,6 +390,49 @@ public void testGroupBy() {
     Assert.assertEquals(asSet("1a#2", "3foobar#6", "5#10"), toSet(cogrouped.collectAsList()));
   }
 
+  @Test
+  public void testMelt() {
```

Comment:
I found it always useful to have these tests as they actually call into the Scala API and proof that this works and is Java-friendly. Otherwise no Java code uses the method and only users find out it is useless. But the `unpivot` method signatures look simple and Java-friendly (and tests have proven they are alright), so I can remove these again.